index.html

```diff
@@ -586,17 +606,72 @@ <h2>
         with the <a>default display mode</a> (<code>browser</code>) being the
         last item in the chain.
       </p>
+      <p>
+        For more advanced usages, the <a data-link-for="WebAppManifest">
+        display_override</a> member can be used to specify a custom fallback
+        order. This member is valid IFF <a data-link-for="WebAppManifest">display</a> is also present.
+      </p>
+      <p>
+        The algorithm for determing the web app's requested display mode is:
+        <ol>
+          <li>[=list/For each=] <var>considering_display_mode</var> of <var>
+            <a data-link-for="WebAppManifest">display_override</a></var>:
+            <ol>
+              <li>
+                If the user agent supports the |considering_display_mode|,
+                then that display mode is used (exit).
+              </li>
+            </ol>
+          </li>
+          <li>
+            Let |considering_display_mode| be the the value specified by the
+            <a data-link-for="WebAppManifest">display</a> property.
+          </li>
+          <li>
+            If the user agent supports the |considering_display_mode|, then
+            that display mode is used (exit).
+          </li>
+          <li>
+            While |considering_display_mode| has <a>fallback display mode</a>,
```

Comment:
I'd like to find a way to express this in a normal while loop rather than effectively using a goto.

I think the right approach here is to essentially allow considering display mode to be undefined. Hasty pseudo-code:

1. Let *considering* be the display property.
2. While *considering* is not undefined:
    1. If the user agent supports *considering*, return *considering*.
    2. Set *considering* to the fallback display mode of *considering* (which is undefined if it has none).

Alternatively, you don't really need a condition, since as your note below says, we're guaranteed to terminate with browser. So you could just do:

1. Let *considering* be the display property.
2. While true:*
    1. If the user agent supports *considering*, return *considering*.
    2. Set *considering* to the fallback display mode of *considering*.

\* I'm not sure what the idiomatic way to do a while-true loop is in [Infra](https://infra.spec.whatwg.org/). There doesn't seem to be any special syntax for "Loop until broken" or "Loop forever". (I've been writing Rust code and I quite like that they have a special keyword `loop {}` for this case, to avoid the somewhat ugly `while true {}` in other languages.)